Invalid bean definition with name / Cannot register bean definition

Folgender Fehler tritt im QS/Produktiv-Betrieb grundsätzlich auf. (Ich glaube aber nur dann, wenn mehrere Topics gemultiplext als Input angegeben werden)

```
2023-02-13 09:44:16.530 ERROR --- [   scheduling-1] o.s.cloud.stream.binding.BindingService  : Failed to create consumer binding; retrying in 30 seconds

org.springframework.cloud.stream.binder.BinderException: Exception thrown while starting consumer:
	at org.springframework.cloud.stream.binder.AbstractMessageChannelBinder.doBindConsumer(AbstractMessageChannelBinder.java:528) ~[spring-cloud-stream-4.0.0.jar:4.0.0]
	at org.springframework.cloud.stream.binder.AbstractMessageChannelBinder.doBindConsumer(AbstractMessageChannelBinder.java:103) ~[spring-cloud-stream-4.0.0.jar:4.0.0]
	at org.springframework.cloud.stream.binder.AbstractBinder.bindConsumer(AbstractBinder.java:144) ~[spring-cloud-stream-4.0.0.jar:4.0.0]
	at org.springframework.cloud.stream.binding.BindingService.lambda$rescheduleConsumerBinding$1(BindingService.java:209) ~[spring-cloud-stream-4.0.0.jar:4.0.0]
	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) ~[spring-context-6.0.3.jar:6.0.3]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
Caused by: org.springframework.beans.factory.support.BeanDefinitionOverrideException: Invalid bean definition with name 'kafka-718005357.sink-in-0.errors' defined in null: Cannot register bean definition [Root bean: class [org.springframework.messaging.SubscribableChannel]; scope=; abstract=false; lazyInit=null; autowireMode=0; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=null; factoryMethodName=null; initMethodNames=null; destroyMethodNames=null] for bean 'kafka-718005357.sink-in-0.errors' since there is already [Root bean: class [org.springframework.messaging.SubscribableChannel]; scope=; abstract=false; lazyInit=null; autowireMode=0; dependencyCheck=0; autowireCandidate=true; primary=false; factoryBeanName=null; factoryMethodName=null; initMethodNames=null; destroyMethodNames=null] bound.
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.registerBeanDefinition(DefaultListableBeanFactory.java:1003) ~[spring-beans-6.0.3.jar:6.0.3]
	at org.springframework.context.support.GenericApplicationContext.registerBeanDefinition(GenericApplicationContext.java:350) ~[spring-context-6.0.3.jar:6.0.3]
	at org.springframework.context.support.GenericApplicationContext.registerBean(GenericApplicationContext.java:556) ~[spring-context-6.0.3.jar:6.0.3]
	at org.springframework.cloud.stream.binder.AbstractMessageChannelBinder.registerErrorInfrastructure(AbstractMessageChannelBinder.java:773) ~[spring-cloud-stream-4.0.0.jar:4.0.0]
	at org.springframework.cloud.stream.binder.AbstractMessageChannelBinder.registerErrorInfrastructure(AbstractMessageChannelBinder.java:705) ~[spring-cloud-stream-4.0.0.jar:4.0.0]
	at org.springframework.cloud.stream.binder.kafka.KafkaMessageChannelBinder.createConsumerEndpoint(KafkaMessageChannelBinder.java:716) ~[spring-cloud-stream-binder-kafka-4.0.0.jar:4.0.0]
	at org.springframework.cloud.stream.binder.kafka.KafkaMessageChannelBinder.createConsumerEndpoint(KafkaMessageChannelBinder.java:159) ~[spring-cloud-stream-binder-kafka-4.0.0.jar:4.0.0]
	at org.springframework.cloud.stream.binder.AbstractMessageChannelBinder.doBindConsumer(AbstractMessageChannelBinder.java:474) ~[spring-cloud-stream-4.0.0.jar:4.0.0]
	... 10 common frames omitted
```

--- a/.github/workflows/ci.yaml
+++ b/.github/workflows/ci.yaml
@@ -53,9 +53,6 @@ jobs:
       - name: Checkout code
         uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3
 
-      - name: Validate Gradle wrapper
-        uses: gradle/wrapper-validation-action@55e685c48d84285a5b0418cd094606e199cca3b6 # v1
-
       - name: Set up QEMU
         uses: docker/setup-qemu-action@e81a89b1732b9c48d79cd809d8d81d79c4647a18 # v2
 

--- a/README.md
+++ b/README.md
@@ -34,7 +34,7 @@ The application publishes useful Prometheus metrics at `/actuator/prometheus`. T
 ### Setup Kafka
 
 ```sh
-docker-compose -f docker-compose.dev.yml up
+docker compose -f docker-compose.dev.yml up
 ```
 
 ```sh

--- a/build.gradle
+++ b/build.gradle
@@ -1,5 +1,5 @@
 plugins {
-    id 'org.springframework.boot' version '3.0.1'
+    id 'org.springframework.boot' version '3.0.2'
     id 'io.spring.dependency-management' version '1.1.0'
     id 'java'
     id 'jacoco'
@@ -22,7 +22,7 @@ repositories {
 }
 
 ext {
-    set('springCloudVersion', '2022.0.0')
+    set('springCloudVersion', '2022.0.1')
 }
 
 dependencies {

--- a/docker-compose.dev.yml
+++ b/docker-compose.dev.yml
@@ -43,7 +43,7 @@ services:
       AKHQ_CONFIGURATION: |
         akhq:
           connections:
-            docker-kafka-server:
+            kafka1:
               properties:
                 bootstrap.servers: "kafka1:19092"
     ports:

--- a/k8s/akhq-values.yaml
+++ b/k8s/akhq-values.yaml
@@ -1,0 +1,10 @@
+configuration:
+  akhq:
+    server:
+      access-log:
+        enabled: false
+        name: org.akhq.log.access
+    connections:
+      my-cluster: # url friendly name for the cluster (letter, number, _, -, ... dot are not allowed here)
+        properties: # standard kafka properties (optional)
+          bootstrap.servers: "my-cluster-kafka-bootstrap:9092"

--- a/k8s/stream-processors-values.yaml
+++ b/k8s/stream-processors-values.yaml
@@ -15,9 +15,9 @@ processors:
         - name: DISABLE_AUTO_COMMIT
           value: "false"
         - name: TOPIC
-          value: fhir-msg
+          value: fhir-msg,fhir-msg-1
         - name: JAVA_TOOL_OPTIONS
-          value: "-XX:MaxRAMPercentage=50 -XX:G1PeriodicGCInterval=1000 -XX:G1PeriodicGCSystemLoadThreshold=1000"
+          value: "-XX:MaxRAMPercentage=70"
       resources:
         requests:
           memory: "1Gi"

--- a/skaffold.yaml
+++ b/skaffold.yaml
@@ -20,6 +20,14 @@ build:
 manifests:
   helm:
     releases:
+      - name: akhq
+        remoteChart: akhq
+        repo: https://akhq.io/
+        version: 0.3.1
+        skipBuildDependencies: true
+        wait: true
+        valuesFiles:
+          - "k8s/akhq-values.yaml"
       - name: hapi-fhir-jpaserver
         remoteChart: hapi-fhir-jpaserver
         repo: https://hapifhir.github.io/hapi-fhir-jpaserver-starter

